### ISSUE TITLE
Advanced Playlist - custom configuration not working

### ISSUE THREAD

**Issue by Hyperjase74** [AzuraCast/AzuraCast#4208]
**Using Docker installation method?**
Yes

**AzuraCast version:**
 v0.12.4 Stable

**Host Operating System:**
Ubuntu 20.04 LTS

**Describe the bug:**
Following the instructions provided on this page; https://docs.azuracast.com/en/user-guide/playlists/advanced-playlists - entering the information as per my configuration (playlist_ads, playlist_jingles etc) under the following setting has no effect - nothing plays, all the playlists are set to Advanced.

**Steps to reproduce:**

1. Go to station page
2. Click Profile
3. Click Edit Profile
4. Select Broadcasting tab
5. Find Advanced Custom Configuration box
6. Enter **radio = rotate(weights=[2,5,1], [playlist_ads,playlist_jingles, playlist_music])**

**Expected behavior:**
The rule provided (with some adjustment) should allow custom playlists, with rotation rules applied, but nothing plays.

**Relevant logs:**
2021/05/24 19:59:41 >>> LOG START
2021/05/24 19:59:40 [main:3] Liquidsoap 1.4.4
2021/05/24 19:59:40 [main:3] Using: bytes=[distributed with OCaml 4.02 or above] pcre=7.4.6 sedlex=2.3 menhirLib=20210310 dtools=0.4.2 duppy=0.8.0 cry=0.6.1 mm=0.5.1 ogg=0.5.2 vorbis=0.7.1 opus=0.1.3 mad=0.4.5 flac=0.1.5 flac.ogg=0.1.5 dynlink=[distributed with Ocaml] lame=0.3.3 fdkaac=0.3.1 ffmpeg=0.4.3 samplerate=0.1.4 taglib=0.3.3 ssl=0.5.9 camomile=1.0.2 faad=0.4.0 ladspa=0.1.5
2021/05/24 19:59:41 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz master.
2021/05/24 19:59:41 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2021/05/24 19:59:41 [frame:3] Targetting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2021/05/24 19:59:41 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2021/05/24 19:59:41 [decoder:3] Method "FFMPEG" accepted "/usr/local/share/icecast/web/error.mp3".
2021/05/24 19:59:41 [sandbox:3] Running inside a docker container, disabling sandboxing..
2021/05/24 19:59:41 [video.converter:3] Couldn't find preferred video converter: gavl.
2021/05/24 19:59:41 [audio.converter:3] Using samplerate converter: ffmpeg.
2021/05/24 19:59:41 [harbor:3] Adding mountpoint '/' on port 8005
2021/05/24 19:59:41 [playlist_default:3] Loading playlist...
2021/05/24 19:59:41 [playlist_default:3] No mime type specified, trying autodetection.
2021/05/24 19:59:41 [playlist_default:3] Playlist treated as format application/x-mpegURL
2021/05/24 19:59:41 [playlist_default:3] Successfully loaded a playlist of 0 tracks.
2021/05/24 19:59:41 [error_jingle:3] "/usr/local/share/icecast/web/error.mp3" is static, resolving once for all...
2021/05/24 19:59:41 [local_1:3] Connecting mount /radio.mp3 for source@127.0.0.1...
2021/05/24 19:59:41 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:41 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:41 [local_1:3] Connection setup was successful.
2021/05/24 19:59:41 [clock.wallclock_main:3] Streaming loop starts, synchronized with wallclock.
2021/05/24 19:59:41 [safe_fallback:3] Switch to amplify_38880.
2021/05/24 19:59:41 [live_fallback:3] Switch to cross_38850.
2021/05/24 19:59:41 [requests_fallback:3] Switch to autodj_fallback.
2021/05/24 19:59:41 [autodj_fallback:3] Switch to dynamic_startup.
2021/05/24 19:59:41 [dynamic_startup:3] Switch to switch_38832.
2021/05/24 19:59:41 [switch_38832:3] Switch to autodj_startup_blank.
2021/05/24 19:59:43 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:43 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:45 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:45 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:47 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:48 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:50 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:50 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:52 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:52 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:54 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:54 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:56 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:56 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:59 [lang:3] AzuraCast Raw Response: 
2021/05/24 19:59:59 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:01 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:01 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:03 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:03 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:05 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:05 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:07 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:07 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:10 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:10 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:12 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:12 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:14 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:14 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:16 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:16 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:19 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:19 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:21 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:21 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:23 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:23 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:25 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:25 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:27 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:28 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:30 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:30 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:32 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:32 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:34 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:34 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:36 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:36 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:38 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:39 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:41 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:41 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:43 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:43 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:45 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:45 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:47 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:47 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:50 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:50 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:52 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:52 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:54 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:54 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:56 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:56 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:58 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:00:58 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:01 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:01 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:03 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:03 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:05 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:05 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:07 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:07 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:09 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:10 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:12 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:12 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:14 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:14 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:16 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:16 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:18 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:18 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:21 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:21 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:21 [lang:3] AutoDJ could not be initialized within the specified timeout.
2021/05/24 20:01:23 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:23 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:25 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:25 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:27 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:27 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:29 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:29 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:32 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:32 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:34 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:34 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:35 [cross_38850:3] Analysis: -infdB / -nandB (5.96s / 0.00s)
2021/05/24 20:01:35 [crossfade:3] Old <= medium, new <= medium and |old-new| <= margin.
2021/05/24 20:01:35 [crossfade:3] Old and new source are not too loud and close.
2021/05/24 20:01:35 [crossfade:3] Transition: crossed, fade-in, fade-out.
2021/05/24 20:01:36 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:36 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:38 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:38 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:40 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:41 [lang:3] AzuraCast Raw Response: 
2021/05/24 20:01:41 [error_jingle:3] Prepared "/usr/local/share/icecast/web/error.mp3" (RID 0).
2021/05/24 20:01:41 [safe_fallback:3] Switch to error_jingle with forgetful transition.
2021/05/24 20:01:41 [ffmpeg:3] [mp3float @ 0x7f5ec800fc40] Could not update timestamps for skipped samples.

**Screenshots:**
![image](https://user-images.githubusercontent.com/84740461/119396321-9db27280-bccc-11eb-9cce-21cf4cadf1d8.png)
![image](https://user-images.githubusercontent.com/84740461/119396439-be7ac800-bccc-11eb-8919-cc2992fb80fd.png)

**Comment by Hyperjase74** [comment]
With a lot of messing around, it turns out placement in which custom Liquidsoap configuration input box you use is vitally important!

I've moved my existing code from the very last input box on the Edit Liquidsoap Configuration page to the box just after this line;

**radio = fallback(id="requests_fallback", track_sensitive = true, [requests, radio])
add_skip_command(radio)**

And now it magically works!

**Comment by silvaalex2021** [comment]
Just commenting, since you found the answer for yourself. You're right! the last box should only be used to create new streams, for example to broadcast to a remote server. I couldn't get any other code to work when placed there!

maybe the document you mentioned on the issue is just a little outdated.

**Comment by Hyperjase74** [comment]
@silvaalex2021 yes, slightly frustrating as the documentation actually gives you the wrong information (as to where to put the code).

I've found another issue from this - even though I have specified the rotation (rotate(weights=[2,1,2], [playlist_music,playlist_jingles,playlist_ads])) - annoyingly, for whatever reason the playlist_jingles (which is set to 1) sometimes plays 2!  It seems to decide to itself to play 2 or 1 ... can't pin down what causes it.
